### DISCUSSION TITLE
Welcome to WGDashboard Discussions!

### DISCUSSION THREAD

**Discussion by donaldzou** [WGDashboard/WGDashboard#300] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by barryboom** [comment]
i would like to have the ability to setup 2FA even if i skip at time of initial setup.

**Reply by donaldzou** [reply]
Yes this is possible now with the latest `v4` branch commit :)

**Reply by donaldzou** [reply]
It is under the dashboard setting

**Comment by Fayyaadh** [comment]
Is it possible to install this alongside an existing Wireguard setup?

I've got Wireguard running on a VM on Proxmox, but I want to add a dashboard to manage clients, without having to redo any setup.

**Reply by donaldzou** [reply]
Hi! Yes that would be possible. WGDashboard will scan existing configurations and you can control them in it :)

**Reply by Fayyaadh** [reply]
> Hi! Yes that would be possible. WGDashboard will scan existing configurations and you can control them in it :)

And for it to integrate with an existing Wireguard installation, I just install WGDashboard as normal per the instructions, nothing special?

**Reply by donaldzou** [reply]
Yes, that is correct. Basically WGDashboard will look for configuration under `/etc/wireguard` by default, and control them with `wg` and `wg-quick`. It doesn't write to the configuration file directly, unless you're creating one :)

**Reply by Fayyaadh** [reply]
> Yes, that is correct. Basically WGDashboard will look for configuration under `/etc/wireguard` by default, and control them with `wg` and `wg-quick`. It doesn't write to the configuration file directly, unless you're creating one :)

And it'll monitor the existing wg0 interface, for stats and last handshake etc, correct?

**Reply by donaldzou** [reply]
That's correct :) U can checkout the screenshots at the end of the readme.md :)

**Reply by Fayyaadh** [reply]
> That's correct :) U can checkout the screenshots at the end of the readme.md :)

This is perfect then! I will try it out later today and give feedback, thank you sir.

**Reply by Fayyaadh** [reply]
I followed the instructions but when trying to start wgdashboard, I get this:

```
sudo ./wgd.sh start
trap: usage: trap [-lp] [[arg] signal_spec ...]
------------------------------------------------------------
[WGDashboard] Starting WGDashboard with Gunicorn in the background.
./wgd.sh: line 59: ./venv/bin/activate: No such file or directory
sudo: ./venv/bin/gunicorn: command not found
```

**Reply by donaldzou** [reply]
Which command did you used to install and which OS are you running on?

**Reply by Fayyaadh** [reply]
> Which command did you used to install and which OS are you running on?

I followed the instructions under the "Manual" installation steps on the main page.

`sudo ./wgd.sh install`

Running on an Ubuntu 22.04 LTS VM..

**Reply by Fayyaadh** [reply]
Hi, any ideas?

I can't get it to install.

**Reply by Dafrazz** [reply]
Same problem here running Debian GNU/Linux 11 (bullseye) x86_64 LXC in Proxmox.
Updated to python 3.10 and still get the same error as OP.

```
sudo ./wgd.sh start
trap: usage: trap [-lp] [[arg] signal_spec ...]
------------------------------------------------------------
[WGDashboard] Starting WGDashboard with Gunicorn in the background.
./wgd.sh: line 59: ./venv/bin/activate: No such file or directory
sudo: ./venv/bin/gunicorn: command not found
```
Any ideas?
Thanks in advance.

**Reply by Fayyaadh** [reply]
> Same problem here running Debian GNU/Linux 11 (bullseye) x86_64 VM in Proxmox. Updated to python 3.10 and still get the same error as OP.
> 
> ```
> sudo ./wgd.sh start
> trap: usage: trap [-lp] [[arg] signal_spec ...]
> ------------------------------------------------------------
> [WGDashboard] Starting WGDashboard with Gunicorn in the background.
> ./wgd.sh: line 59: ./venv/bin/activate: No such file or directory
> sudo: ./venv/bin/gunicorn: command not found
> ```
> 
> Any ideas? Thanks in advance.

Yeah, I gave up until I get some guidance on what to try from the maintainer.

I googled the error and tried some suggestions but none worked for me

**Reply by Dafrazz** [reply]
Same ran around in circles from updateing python and trying everything on instal page for Debian 11 and the manual install, still same result, Quite frustrating. Guess im leaving it for a while until its figured out what i did wrong.

**Reply by Fayyaadh** [reply]
> Same ran around in circles from updateing python and trying everything on instal page for Debian 11 and the manual install, still same result, Quite frustrating. Guess im leaving it for a while until its figured out what i did wrong.

I was looking to add a nice dashboard to my existing Wireguard VM in Proxmox and this repo seemed very promising. Hopefully, the maintainer sees this discussion and helps out.

**Reply by Dafrazz** [reply]
> > Same ran around in circles from updateing python and trying everything on instal page for Debian 11 and the manual install, still same result, Quite frustrating. Guess im leaving it for a while until its figured out what i did wrong.
> 
> I was looking to add a nice dashboard to my existing Wireguard VM in Proxmox and this repo seemed very promising. Hopefully, the maintainer sees this discussion and helps out.

I think i may have figured it out. It seems (in my case ) the default install is under the
 /root/WGDashboard 
 directory not 
 /etc/wgdashboard
 
 I changed the directory did ( as per the manual install instructed )
 
 ` cd ./WGDashboard/src 
 
chmod +x ./wgd.sh 
 
./wgd.sh install 
 
sudo echo "net.ipv4.ip_forward=1" >> /etc/sysctl.conf 

sudo sysctl -p /etc/sysctl.conf `

`./wgd.sh start`

All went well, it started i could login and see the UI....

Success!!!

 I have other problems though, maybe its a permission thing but the peers section isnt reporting the configs under wireguard default connection  wg0 or the peers config files.
 
 When i try to change the peer to the settings in the UI under /etc/wireguard/configs it wont accept my DDNS name.
 
 No matter though , I just cant change the peer info on the dashboard, The names come up as "Untitled", not a game breaker as wireguard access all still works fine.
 
 I will carry on manually using the CLI to add peers under pivpn -a rather than the UI
 
  One step closer.

***Edited wrong file path***

**Comment by quando9046** [comment]
Hello. Will refreshing the user work? In version 4. when connecting to the server the green light comes on and everything is ok but when I turn off the client it still shows that it is connected?

**Reply by donaldzou** [reply]
Ahhh I never thought about this circumstances. I will fix it later today

**Reply by quando9046** [reply]
Thank you and please forgive me but it is quite important. Good job :))

**Reply by donaldzou** [reply]
Ohh wait... I completely understood your question wrong.. yes the green light will be off if the last handshake time is more than 2 minutes. It can't be accurate due to the nature of UDP, unlike TCP

**Reply by quando9046** [reply]
well, it's probably not good because 4 minutes have passed and it still won't turn on. :((

**Reply by donaldzou** [reply]
> well, it's probably not good because 4 minutes have passed and it still won't turn on. :((

What do you mean by won't turn on?

**Reply by quando9046** [reply]
green light showing the user's connection. When the user connects to the server, it should light up green and when they disconnect from the server, it should go out

**Reply by donaldzou** [reply]
Do you mind providing a screenshot please?

**Reply by quando9046** [reply]
![Bez tytułu](https://github.com/user-attachments/assets/dedc9fbf-eeca-4040-9ba2-7cb9bc3151d7)
see user test1_wg1 is disabled and on the server it shows that he is active and this does not change

**Reply by donaldzou** [reply]
Gotcha.. I've pushed an update to the code and it should fix the issue now.

To update, please view: [How to update](https://github.com/donaldzou/WGDashboard?tab=readme-ov-file#-how-to-update-the-dashboard)

**Reply by quando9046** [reply]
Great. Now everything works fine. Thank you.

**Reply by quando9046** [reply]
I have one more question about the wg0.conf autostart. During the server restart it does not start automatically. I have to start it manually. If I restart it with the command ./wgd.sh restart in the terminal, everything works properly, wg0.conf is started. A full server restart does not start wg0.conf. Please understand and thank you

Sorry but now I read that this function is in the backlog to do. It remains to wait for the update and implementation of this function.

**Comment by ElkyBoy** [comment]
Hello everyone! Will this dashboard conflict with PiHole web interface (lightppd)?

**Comment by MrAuger** [comment]
Hello, guys.
How to restrict the download speed for any client?

**Reply by donaldzou** [reply]
As far as I know.. is hard to limit a specific peer, but you can limit the whole WG configuration with linux `tc` commands.

**Reply by MrAuger** [reply]
No, thanks. THere is no a good idea to limit the bandwidth for all clients... Only for some of them.

**Reply by MrAuger** [reply]
# Here the answer:

1. tc qdisc add dev ens3 root handle 1:0 hfsc default 1
2. tc class add dev ens3 parent 1:0 classid 1:1 hfsc sc rate 100mbit ul rate 100mbit
3. tc class add dev ens3 parent 1:0 classid 1:2 hfsc sc rate 1000kbit ul rate 1000kbit
4. tc filter add dev ens3 protocol all parent 1: u32 match ip dst 10.66.66.9 flowid 1:2
